### PR TITLE
Rewrite deploy script

### LINKED ISSUE #781: Fix Release Script
- [ ] Separate to 3 parts like Vega 
  - npm release (old code should work)
  -  [ ] bower release should not include bower_components folder (Current it does include the old  `gallery/bower_components`)
  - [ ] Github pages release should include bower_components

### LINKED ISSUE #840: Gallery broken in deployment broken because of symlinks

### PR DESCRIPTION
**Pull request by kanitw**
Fixes #781, Fixes #840 

- Separate deploy script into 3 parts like Vega 
  - npm release (old code should work)
  - bower release, which does not include bower_components folder
  - github pages release `deploy-gh.sh`, which include bower_components

`deploy-gh.sh` is well tested.  See http://vega.github.io/vega-lite/gallery.html

Other parts will be further tested when we release `0.9.1`.

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -1,29 +1,33 @@
-set -e
-
-# define color
-RED='\033[0;31m'
-NC='\033[0m' # No Color
+. ./scripts/pre-deploy.sh
 
+# 1. NPM PUBLISH
```
Comment (domoritz) [inline scripts/deploy.sh]: Why did you move npm first?

File: package.json [hunk 2]
```diff
@@ -15,12 +15,15 @@
   "scripts": {
     "build": "browserify src/vl.ts -p tsify -d -s vl | exorcist vega-lite.js.map > vega-lite.js",
     "postbuild": "uglifyjs vega-lite.js -cm --source-map vega-lite.min.js.map > vega-lite.min.js && npm run schema",
+    "build:all": "npm run clean && npm run build && npm run schema:only && npm test && npm run lint",
     "cover": "tsc && istanbul cover node_modules/.bin/_mocha -- --recursive",
     "clean": "rm -f vega-lite.* vega-lite-schema.json & find src -name '*.js*' -type f -delete & find test -name '*.js*' -type f -delete & rm -rf examples/_diff examples/_original examples/_output",
-    "deploy": "npm run clean && npm run lint && npm run test && scripts/deploy.sh",
+    "deploy": "scripts/deploy.sh",
+    "deploy-gh": "scripts/deploy-gh.sh",
```
Comment (domoritz) [inline package.json]: deploy:gh

File: scripts/deploy.sh (hunk 1 above)
Comment (kanitw) [inline scripts/deploy.sh]: just for consistency with Vega.
